### PR TITLE
MES-4442: Cat A Mod 2 Rekey Reason pages

### PR DESCRIPTION
**Pull request by MarkPatt89**
## Description
- Cat A Mod 2 Rekey Reason pages added.
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [x] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/page-names.constants.ts [hunk 1]
```diff
@@ -14,6 +14,10 @@ export const CAT_A_MOD1 = {
   REKEY_UPLOAD_OUTCOME_PAGE: 'RekeyUploadOutcomePage',
 };
 
+export const CAT_A_MOD2 = {
+  REKEY_UPLOAD_OUTCOME_PAGE: 'RekeyUploadOutcomePage',
```
Comment (amobarak) [inline src/pages/page-names.constants.ts:18]: Should this be `RekeyReasonCatAMod2Page `

Comment (MarkPatt89) [reply src/pages/page-names.constants.ts:18]: No, I don't believe so. This page does not get refactored for each category.

**Review by amobarak — APPROVED** [review]

File: src/pages/rekey-reason/cat-a-mod2/__tests__/rekey-reason.cat-a-mod2.page.spec.ts [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+      });
+      it('should navigate to the next page and not display the retry modal when an upload is a duplicate', () => {
+        const action = new SendCurrentTestFailure(true);
+        const result: RekeyReasonModel = rekeyReasonReducer(null, action);
+        const uploadStatus = getUploadStatus(result);
+
+        component.handleUploadOutcome(uploadStatus);
+
+        expect(component.handleLoadingUI).toHaveBeenCalledWith(false);
+        expect(navContoller.push).toHaveBeenCalledWith(CAT_A_MOD1.REKEY_UPLOAD_OUTCOME_PAGE);
```
Comment (appleabs) [inline src/pages/rekey-reason/cat-a-mod2/__tests__/rekey-reason.cat-a-mod2.page.spec.ts]: should this be mod2 not mod1?

File: src/pages/rekey-reason/cat-a-mod2/__tests__/rekey-reason.cat-a-mod2.page.spec.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+      });
+      it('should navigate to next page and not display the retry modal when an upload succeeds', () => {
+        const action = new SendCurrentTestSuccess();
+        const result: RekeyReasonModel = rekeyReasonReducer(null, action);
+        const uploadStatus = getUploadStatus(result);
+
+        component.handleUploadOutcome(uploadStatus);
+
+        expect(component.handleLoadingUI).toHaveBeenCalledWith(false);
+        expect(navContoller.push).toHaveBeenCalledWith(CAT_A_MOD1.REKEY_UPLOAD_OUTCOME_PAGE);
```
Comment (appleabs) [inline src/pages/rekey-reason/cat-a-mod2/__tests__/rekey-reason.cat-a-mod2.page.spec.ts]: Same as previous, should this be Mod2?

**Review by szabikr — APPROVED** [review]

**Review by appleabs — APPROVED** [review]

**Review by TomBillingtonUK — APPROVED** [review]
Looks good

File: src/pages/rekey-reason/cat-a-mod2/__tests__/rekey-reason.cat-a-mod2.page.spec.ts (hunk 2 above)
Comment (MarkPatt89) [reply src/pages/rekey-reason/cat-a-mod2/__tests__/rekey-reason.cat-a-mod2.page.spec.ts]: Thanks Jen, you're right.

File: src/pages/rekey-reason/cat-a-mod2/__tests__/rekey-reason.cat-a-mod2.page.spec.ts (hunk 3 above)
Comment (MarkPatt89) [reply src/pages/rekey-reason/cat-a-mod2/__tests__/rekey-reason.cat-a-mod2.page.spec.ts]: And here, also. :)

**Pull request merged**